### PR TITLE
Integrate supabase

### PR DESCRIPTION
**Pull request by LucasXu0**
### Feature Preview

---

#### PR Checklist

- [ ] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/user/application/auth/appflowy_auth_service.dart [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import '../../../generated/locale_keys.g.dart';
+
+class AppFlowyAuthService implements AuthService {
+  @override
+  Future<Either<FlowyError, UserProfilePB>> signIn({
+    required String email,
+    required String password,
+  }) async {
+    //
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/user/application/auth/appflowy_auth_service.dart:19]: Remove

File: frontend/appflowy_flutter/lib/user/application/auth/appflowy_auth_service.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    final request = SignInPayloadPB.create()
+      ..email = email
+      ..password = password;
+
+    final response = UserEventSignIn(request).send();
+    return response.then((value) => value.swap());
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/user/application/auth/appflowy_auth_service.dart:25]:
```suggestion
    final response = await UserEventSignIn(request).send();
    return response.swap());
```

File: frontend/appflowy_flutter/lib/user/application/auth/appflowy_auth_service.dart [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    final response = UserEventSignUp(request).send();
+    return response.then((value) => value.swap());
+  }
+
+  @override
+  Future<void> signOut() async {
+    await UserEventSignOut().send();
+    return;
+  }
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/user/application/auth/appflowy_auth_service.dart:47]:
```suggestion
  Future<void> signOut() async => await UserEventSignOut().send();
```

File: frontend/appflowy_flutter/lib/user/application/auth/auth_service.dart [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  Future<Either<FlowyError, UserProfilePB>> signUp({
+    required String name,
+    required String email,
+    required String password,
+  });
+
+  /// Returns a default [UserProfilePB]
+  Future<Either<FlowyError, UserProfilePB>> signUpAsAnonymousUser();
+
+  ///
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/user/application/auth/auth_service.dart:22]:
```suggestion
  /// Signs out the currently signed-in user.
```

File: frontend/appflowy_flutter/lib/user/application/auth/supabase_auth_service.dart [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    required String email,
+    required String password,
+  }) async {
+    try {
+      final user = await _auth
+          .signInWithPassword(
+            email: email,
+            password: password,
+          )
+          .then((value) => value.user);
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/user/application/auth/supabase_auth_service.dart:42]:
```suggestion
      final response = await _auth.signInWithPassword(
        email: email,
        password: password,
      );
      
      final user = response.user;
```
Again, just personal preference, doesn't actually matter..

File: frontend/appflowy_flutter/lib/user/application/auth/appflowy_auth_service.dart (hunk 2 above)
Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/user/application/auth/appflowy_auth_service.dart:27]: both of them look good to me. `await` or `then`. 

Sure, we can generate the swap function while converting the PB event to a function.

**Comment by LucasXu0** [comment]
1. Add a `.env` file in the `appflowy_flutter` directory.
2. Run `flutter pub run build_runner build --delete-conflicting-outputs` to generate the keys from the env file.
3. If you want to regenerate the keys, you need to run `flutter pub run build_runner clean` before running `flutter pub run build_runner build --delete-conflicting-outputs`.

Sample for `.env` file
```
SUPABASE_URL=https://**********.supabase.co
SUPABASE_ANON_KEY=wgsdf**********gewr
SUPABASE_JWT_SECRET=sdfggh**********wervvdr==
```

**Comment by Xazin** [comment]
Is envied better than flutter_dotenv?

Anholdt, might be good to add a `.env.example` that contains all keys. Maybe mark required ones with ** or similar.

**Comment by LucasXu0** [comment]
> Is envied better than flutter_dotenv?
> 
> Anyhow, might be good to add a `.env.example` that contains all keys. Maybe mark required ones with ** or similar.

Sure, we should include a `.env.example` file as an example. However, using the flutter_dotenv package requires adding the env file to the assets, which may not be secure for storing keys.

**Pull request merged**